spec/models/change_service_event_spec.rb

```diff
@@ -21,6 +21,27 @@
       expect(change_service_event.data['name']).to eql(new_service_name)
     end
   end
+  context '#trigger_publish_event' do
+    it 'when entity id is changed on an sp_component' do
+      event = create(:assign_sp_component_to_service_event, service: service)
+      updated_entity_id = 'https://changed-entity-id1'
+      service.assign_attributes(entity_id: updated_entity_id)
+      change_service_event = create(:change_service_event, service: service)
+
+      resulting_event = PublishServicesMetadataEvent.where("data->>'event_id' = ?", change_service_event.id).first
+      expect(resulting_event).to be_present
+      expect(resulting_event.data['services_metadata']['connected_services'][0]['entity_id']).to eq(updated_entity_id)
+    end
+    it 'when entity id is changed on an msa_component' do
+      event = create(:assign_msa_component_to_service_event, service: service)
+      updated_entity_id = 'https://changed-entity-id1'
+      service.assign_attributes(entity_id: updated_entity_id)
+      change_service_event = create(:change_service_event, service: service)
+
+      resulting_event = PublishServicesMetadataEvent.where("data->>'event_id' = ?", change_service_event.id).first
+      expect(resulting_event).to be_present
```

Comment:
I think not nil reads better, than using present

the expectations should be consistent across the tests

Reply:
Changed it.. please approve :lion: